fvm/src/call_manager/default.rs

```diff
@@ -471,6 +477,153 @@ where
         })
     }
 
+    fn tst_validate(&mut self, msg: Message, sig: Signature) -> Result<()> {
+        // Lookup the actor.
+        let state = self
+            .state_tree()
+            .get_actor_id(msg.from.id().unwrap())?
+            .ok_or_else(|| syscall_error!(NotFound; "actor does not exist: {}", msg.from))?;
+
+        // Builtin or external?
+        // TODO
```

Comment:
We will need some form of caching to memoize if a given sender code CID exposes a validate entrypoint or not. That way, we don't have to bootstrap an entire Wasm instance only to find out that the actor couldn't act as a sender. Could you open an issue to track this, and maybe jot down some ideas?

Reply:
I think an open issue for this, though it will probably end up being a new field added to `ActorState` to be checked when calling `.get_actor()` though this may be a good place to cache calls into `upgrade` as well.